Upload-Wheels.sh

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-cd download || exit
+cd downloaded_wheels || exit
```

Comment:
Maybe `exit 1` here ... (avoid plain exit).

This script as written now is kind of unreadable and also without handling possible errors.
As minimum, I would suggest check this script with `shellcheck`.
***
Maybe it could be rewritten to make it more readable and robust:
```bash
#!/usr/bin/env bash

usage() {
  echo "Usage: $0 <s3-bucket-name>"
  echo "This script uploads .whl files from the downloaded_wheels directory to the specified S3 bucket."
  exit 1
}

# Validate input
if [ -z "$1" ]; then
  echo "Error: S3 bucket name not provided."
  usage
fi

# Check if AWS CLI installed
if ! command -v aws &> /dev/null; then
  echo "Error: AWS CLI is not installed."
  exit 1
fi

# Change to downloaded_wheels directory
cd downloaded_wheels || { echo "Error: downloaded_wheels directory not found."; exit 1; }

# Check if there are any .whl files
if [ -z "$(ls *.whl 2>/dev/null)" ]; then
  echo "No .whl files found in downloaded_wheels directory."
  exit 1
fi

# Loop over .whl files
for wfile in *.whl; do
  # Process file name
  file_name="${wfile%%-*}"
  file_name="${file_name,,}"  # Convert to lowercase
  file_name="${file_name//_/-}"  # Replace underscores with hyphens

  # Skip esptool for specific OS
  case "$(uname -sr)" in
    CYGWIN*|MINGW*|MINGW32*|MSYS*)
      if [[ "$file_name" == "esptool" ]]; then
        continue
      fi
      ;;
    *)
      ;;
  esac

  # Upload to S3
  if ! aws s3 cp --acl=public-read --no-progress "$wfile" "s3://$1/pypi/$file_name/$wfile"; then
    echo "Error uploading $wfile to S3."
    exit 1
  fi
done

echo "All good ... all files uploaded successfully!"

```

(In case of time pressure, you can ignore this comment for now - just a suggestion.)